### PR TITLE
Added eslint and did lots of linting

### PR DESCRIPTION
**Pull request by christopherhafke**
Properly configured eslint and fixed those things that couldn't automatically be fixed.

Should be a refactor only with no change to behavior.

Also, runs the linter on Travis.

Also, the diff is messed up since I accidentally did a push onto dev which I had to revert, lol.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,4 +11,6 @@ and get going!
 
 To run tests, run `yarn test`.
 
+To lint (i.e. check for style, etc.), run `yarn lint`, and to lint and automatically repair some issues, run `yarn lint-fix`.
```
Comment (eashwar) [inline README.md:14]: image

File: yarn.lock [hunk 2]
```diff
@@ -971,7 +971,7 @@
   dependencies:
     regenerator-runtime "^0.12.0"
 
-"@babel/runtime@^7.1.2":
+"@babel/runtime@^7.1.2", "@babel/runtime@^7.4.5":
```
Comment (eashwar) [inline yarn.lock:878]: This being a structural change with library versions being bumped, perhaps this PR can incorporate #37 and #21 and then we could close those two?

File: package.json [hunk 3]
```diff
@@ -26,7 +27,8 @@
     "build": "react-scripts test --watchAll=false --env=jsdom; react-scripts build",
     "test": "react-scripts test --env=jsdom",
     "eject": "react-scripts eject",
-    "lint": "eslint --ext .jsx,.js src"
+    "lint": "eslint --ext .jsx,.js src",
+    "lint-fix": "eslint --fix --ext .jsx,.js src"
```
Comment (eashwar) [inline package.json:31]: perhaps rename this to `yarn style`, `style` being the imperative verb form for 'resolve style issues'

**Review by eashwar — APPROVED** [review]

**Review by christopherhafke — COMMENTED** [review]
LGTM!

**Pull request merged**